lms/static/scripts/frontend_apps/components/dashboard/StudentsActivity.tsx

```diff
@@ -6,33 +6,32 @@ import {
 } from '@hypothesis/frontend-shared';
 import type { DataTableProps } from '@hypothesis/frontend-shared';
 import { useState } from 'preact/hooks';
+import { useParams } from 'wouter-preact';
 
-import type { StudentStats } from '../../api-types';
+import type { APIAssignment, StudentStats } from '../../api-types';
+import { useConfig } from '../../config';
+import { useAPIFetch } from '../../utils/api';
 import { formatDateTime } from '../../utils/date';
 
-export type AssignmentInfo = {
-  title: string;
-};
-
-export type StudentsActivityTableProps = {
-  assignment: AssignmentInfo;
-  students: StudentStats[];
-  loading?: boolean;
-};
-
 type MandatoryOrder<T> = NonNullable<DataTableProps<T>['order']>;
 
-export default function StudentsActivityTable({
-  assignment,
-  students,
-  loading,
-}: StudentsActivityTableProps) {
-  const title = `Assignment: ${assignment.title}`;
+export default function StudentsActivity() {
+  const { dashboard } = useConfig(['dashboard']);
+  const { links: routes } = dashboard;
+  const { assignmentId } = useParams<{ assignmentId: string }>();
+  const assignment = useAPIFetch<APIAssignment>(
+    routes.assignmentApi.replace(':id_', assignmentId),
+  );
+  const students = useAPIFetch<StudentStats[]>(
+    routes.assignmentStatsApi.replace(':id_', assignmentId),
+  );
```

Comment:
We should bring the logic used in client to dynamically replace params in template URLs, rather than doing a simple `String.replace`.